processor/src/chiplets/hasher/mod.rs

```diff
@@ -25,6 +25,16 @@ pub use aux_trace::{AuxTraceBuilder, SiblingTableRow, SiblingTableUpdate};
 #[cfg(test)]
 mod tests;
 
+/// Result of a merkle tree node update. The result contains the old merkle_root, which
+/// corresponding to the old_value, and the new merkle_root, for the updated value. As well as the
+/// row address of the execution trace at which the computation started.
+#[derive(Debug, Copy, Clone)]
+pub struct MerkleRootUpdate {
```

Comment:
nit: we should probably declare this helper struct inside the `super`module as it will be exposed by the public API and consumed by the children of `chiplets`.

Also, we put a `// HELPER STRUCTS` header for such cases.